tests/lint/unittest_lint.py

```diff
@@ -170,6 +171,16 @@ def test_more_args(fake_path, case):
         assert sys.path == fake_path
 
 
+def test_namespace_package_sys_path() -> None:
+    """Test that we do not append namespace packages to sys.path."""
+    result = subprocess.run(
+        [sys.executable, "-m", "pylint", "test_namespace_pkg/"],
+        check=False,
+        stdout=subprocess.PIPE,
+    )
+    assert "Module import itself" not in result.stdout.decode("utf-8")
```

Comment:
I'd prefer to demo in the tests what our docs say to do. (`subprocess` seems like something we may as well avoid here.)
```suggestion
    pylint_output = StringIO()
    reporter = text.TextReporter(pylint_output)
    Run(["test_namespace_pkg/"], reporter=reporter, exit=False)
    assert "Module import itself" not in pylint_output.getvalue()
```